app/controllers/report_generator_controller.rb

```diff
@@ -0,0 +1,27 @@
+class ReportGeneratorController < ApplicationController
```

Comment:
You will want to be sure to inherit from `AdminAreaController` so that only admins can access this page